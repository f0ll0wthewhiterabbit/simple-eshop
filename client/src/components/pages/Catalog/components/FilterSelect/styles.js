import styled from 'styled-components'
import { FormControl } from '@material-ui/core'

export const Wrapper = styled.div`
  position: absolute;
  top: 49px;
  right: 32px;
`

export const SelectFormControl = styled(FormControl)`
  min-width: 60px;
  margin-left: 10px;

  .MuiInputBase-root {
    font-size: 1.2rem;
  }

  .MuiSelect-root {
    padding-left: 8px;
  }
`
