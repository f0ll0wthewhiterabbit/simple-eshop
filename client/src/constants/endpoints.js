export const PRODUCTS_API = '/data/products.json'
export const USERS_API = '/data/users.json'
