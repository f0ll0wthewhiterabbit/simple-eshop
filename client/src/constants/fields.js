export const STORAGE_FIELD_ACCESS_TOKEN = 'accessToken'
export const STORAGE_FIELD_REFRESH_TOKEN = 'refreshToken'
export const STORAGE_FIELD_THEME = 'theme'
export const FIELD_THEME_DEFAULT = 'default'
export const FIELD_THEME_DARK = 'dark'
export const STORE_FIELD_PRODUCTS = 'products'
export const STORE_FIELD_USERS = 'users'
export const STORE_FIELD_CURRENT_USER = 'usersCurrent'
export const URL_FIELD_RATINGS_FILTER = 'myRatings'
export const URL_FIELD_NO_FILTER = 'all'
