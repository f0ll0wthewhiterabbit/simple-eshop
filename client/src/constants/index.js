export * from './pagePaths'
export * from './endpoints'
export * from './fields'
export * from './roles'
export * from './pageLimits'
