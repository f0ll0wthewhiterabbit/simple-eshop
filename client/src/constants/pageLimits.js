export const DEFAULT_CATALOG_PER_PAGE_LIMIT = 9
export const DEFAULT_ADMIN_PER_PAGE_LIMIT = 10
export const ADMIN_LOW_PER_PAGE_LIMIT = 5
export const ADMIN_HIGH_PER_PAGE_LIMIT = 25
