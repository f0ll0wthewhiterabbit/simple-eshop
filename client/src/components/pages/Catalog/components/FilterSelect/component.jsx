import React from 'react'
import PropTypes from 'prop-types'
import { Select, MenuItem } from '@material-ui/core'

import { Wrapper, SelectFormControl } from './styles'
import { URL_FIELD_NO_FILTER, URL_FIELD_RATINGS_FILTER } from '../../../../../constants'

const FilterSelect = ({ filter, fetchProducts, setProductsFilter }) => {
  const handleChange = event => {
    const { value } = event.target
    if (value === filter) {
      return
    }

    setProductsFilter(value)

    fetchProducts(1, null, value)
  }

  return (
    <Wrapper>
      <SelectFormControl>
        <Select value={filter} onChange={handleChange} data-test="select">
          <MenuItem value={URL_FIELD_NO_FILTER}>All</MenuItem>
          <MenuItem value={URL_FIELD_RATINGS_FILTER}>My ratings</MenuItem>
        </Select>
      </SelectFormControl>
    </Wrapper>
  )
}

FilterSelect.propTypes = {
  filter: PropTypes.string.isRequired,
  fetchProducts: PropTypes.func.isRequired,
  setProductsFilter: PropTypes.func.isRequired,
}

export default FilterSelect
