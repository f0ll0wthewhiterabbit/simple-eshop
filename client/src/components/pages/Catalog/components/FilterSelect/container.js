import { connect } from 'react-redux'

import FilterSelect from './component'
import { fetchProducts, setProductsFilter } from '../../../../../store/actions'

const mapStateToProps = state => ({
  filter: state.getIn(['products', 'filter']),
})

const mapDispatchToProps = dispatch => ({
  fetchProducts: (page, itemsPerPage, filter) =>
    dispatch(fetchProducts(page, itemsPerPage, filter)),
  setProductsFilter: filter => dispatch(setProductsFilter(filter)),
})

export default connect(mapStateToProps, mapDispatchToProps)(FilterSelect)
