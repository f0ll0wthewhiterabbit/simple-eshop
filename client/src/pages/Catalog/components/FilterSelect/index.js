import FilterSelect from './container'

export default FilterSelect
