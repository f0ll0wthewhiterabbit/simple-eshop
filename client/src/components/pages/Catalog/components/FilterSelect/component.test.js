import React from 'react'
import { shallow } from 'enzyme'

import FilterSelect from './component'
import findByTestAttr from '../../../../../utils/findByTestAttr'
import { URL_FIELD_NO_FILTER, URL_FIELD_RATINGS_FILTER } from '../../../../../constants'

describe('FilterSelect component', () => {
  let props
  let wrapper
  const mockFetchProducts = jest.fn()
  const mockSetProductsFilter = jest.fn()

  const generateWrapper = passedProps => {
    const initialProps = {
      filter: URL_FIELD_NO_FILTER,
      fetchProducts: mockFetchProducts,
      setProductsFilter: mockSetProductsFilter,
    }
    props = { ...initialProps, ...passedProps }

    return shallow(<FilterSelect {...props} />)
  }

  beforeEach(() => {
    wrapper = generateWrapper()
  })

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render filter from props', () => {
    const select = findByTestAttr(wrapper, 'select')

    expect(select.props().value).toBe(URL_FIELD_NO_FILTER)
  })

  it('should handle filter change', () => {
    const select = findByTestAttr(wrapper, 'select')
    select.simulate('change', { target: { value: URL_FIELD_RATINGS_FILTER } })

    expect(mockSetProductsFilter).toHaveBeenCalledTimes(1)
    expect(mockSetProductsFilter).toHaveBeenCalledWith(URL_FIELD_RATINGS_FILTER)
    expect(mockFetchProducts).toHaveBeenCalledTimes(1)
    expect(mockFetchProducts).toHaveBeenCalledWith(1, null, URL_FIELD_RATINGS_FILTER)
  })
})
