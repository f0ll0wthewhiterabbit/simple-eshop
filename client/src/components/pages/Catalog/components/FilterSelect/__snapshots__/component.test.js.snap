// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSelect component should render correctly 1`] = `
<styled.div>
  <Styled(WithStyles(ForwardRef(FormControl)))>
    <WithStyles(ForwardRef(Select))
      data-test="select"
      onChange={[Function]}
      value="all"
    >
      <WithStyles(ForwardRef(MenuItem))
        value="all"
      >
        All
      </WithStyles(ForwardRef(MenuItem))>
      <WithStyles(ForwardRef(MenuItem))
        value="myRatings"
      >
        My ratings
      </WithStyles(ForwardRef(MenuItem))>
    </WithStyles(ForwardRef(Select))>
  </Styled(WithStyles(ForwardRef(FormControl)))>
</styled.div>
`;
